core/src/main/java/org/apache/calcite/avatica/remote/AvaticaCommonsHttpClientImpl.java

```diff
@@ -109,34 +110,33 @@ protected void initializeClient(PoolingHttpClientConnectionManager pool,
     HttpClientBuilder httpClientBuilder = HttpClients.custom().setConnectionManager(pool)
         .setDefaultRequestConfig(requestConfig);
     this.client = httpClientBuilder.build();
+
```

Comment:
have you tried setting these directly on the client object ?

Reply:
Not yet but i will try.

Reply:
I have tried setting these directly on the client object, but I think it would require quite a lot of refactoring. For example, practically the whole AvaticaHttpClientFactoryImpl class should be rewritten.

Reply:
Good point.
Maybe in a followup ticket.